app/org/sagebionetworks/bridge/services/AccountWorkflowService.java

```diff
@@ -168,29 +167,29 @@ public void notifyAccountExists(Study study, Email email) {
      * the email does not map to an account, in order to prevent account enumeration 
      * attacks.
      */
-    public void requestResetPassword(Study study, Email email) {
-        checkNotNull(study);
-        checkNotNull(email);
+    public void requestResetPassword(AccountId accountId) {
+        checkNotNull(accountId);
         
-        Account account = accountDao.getAccountWithEmail(study, email.getEmail());
+        Account account = accountDao.getAccount(accountId);
         if (account != null) {
-            sendPasswordResetRelatedEmail(study, email, study.getResetPasswordTemplate());    
+            Study study = studyService.getStudy(account.getStudyIdentifier());
+            sendPasswordResetRelatedEmail(study, account.getEmail(), study.getResetPasswordTemplate());    
```

Comment:
It looks like in the old code, we already take in study. Would it be better to keep that, instead of calling studyService to get the study again?

Reply:
Okay, carried study over from services. Added a precondition check that study ID and account ID are the same.